tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -86,15 +99,22 @@ private void Run(BsonDocument shared, BsonDocument test)
             DropCollection();
             CreateCollection();
             InsertData(shared);
-
+            
             using (ConfigureFailPoint(test))
             {
                 var eventCapturer = new EventCapturer()
                     .Capture<CommandStartedEvent>(e => !__commandsToNotCapture.Contains(e.CommandName));
 
                 Dictionary<string, BsonValue> sessionIdMap;
 
-                using (var client = CreateDisposableClient(test, eventCapturer))
+                var useMultipleShardRouters = shared["_path"].AsString.EndsWith("pin-mongos.json") 
+                    && CoreTestConfiguration.Cluster.Description.Type == ClusterType.Sharded; // could this ever be unknown?
```

Comment:
Could the cluster's type ever be ``Unknown``? I want to say "no", but wouldn't that be be relying on the fact that other tests have been run before it or that a heartbeat has occurred (the latter of which I suppose is a very safe assumption).